src/initialization/MOM_state_initialization.F90

```diff
@@ -1201,6 +1225,84 @@ subroutine trim_for_ice(PF, G, GV, US, ALE_CSp, tv, h, just_read)
 
 end subroutine trim_for_ice
 
+!> Calculate the hydrostatic equilibrium position of the surface under an ice shelf
+subroutine calc_sfc_displacement(PF, G, GV, US, mass_shelf, tv, h)
+  type(param_file_type),   intent(in)    :: PF !< Parameter file structure
+  type(ocean_grid_type),   intent(in)    :: G  !< Ocean grid structure
+  type(verticalGrid_type), intent(in)    :: GV !< Vertical grid structure
+  type(unit_scale_type),   intent(in)    :: US !< A dimensional unit scaling type
+  real, dimension(SZI_(G),SZJ_(G)), &
+                           intent(in)    :: mass_shelf  !< Ice shelf mass [RZ ~> kg m-2]
+  type(thermo_var_ptrs),   intent(inout) :: tv !< Thermodynamics structure
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)), &
+                           intent(inout) :: h  !< Layer thickness [H ~> m or kg m-2]
+
+  real :: z_top_shelf(SZI_(G),SZJ_(G))  ! The depth of the top interface under ice shelves [Z ~> m]
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)+1) :: &
+                                   eta  ! The free surface height that the model should use [Z ~> m].
+  ! temporary arrays
+  real, dimension(SZK_(GV)) :: rho_col   ! potential density in the column for use in ice
+  real, dimension(SZK_(GV)) :: rho_h     ! potential density multiplied by thickness [RZ ~> kg m-2 ]
+  real, dimension(SZK_(GV)) :: h_tmp     ! temporary storage for thicknesses [H ~> m]
+  real, dimension(SZK_(GV)) :: p_ref     ! pressure for density [RZ ~> kg m-2]
+  real, dimension(SZK_(GV)+1) :: ei_tmp, ei_orig ! temporary storage for interface positions [Z ~> m]
+  real :: z_top, z_col, mass_disp, residual, tol
+  integer :: is, ie, js, je, k, nz, i, j, max_iter, iter
+
+  is = G%isc ; ie = G%iec ; js = G%jsc ; je = G%jec ; nz = GV%ke
+
+  tol = US%m_to_Z*0.001 ! 1mm tolerance
+  max_iter = 1e3
+
+  call MOM_mesg("Started calculating initial interface position under ice shelf ")
+  ! Convert thicknesses to interface heights.
+  call find_eta(h, tv, G, GV, US, eta, dZref=G%Z_ref)
+  do j = js, je ; do i = is, ie
+    iter=1
+    z_top_shelf(i,j) = 0.0
+    p_ref(:) = tv%p_ref
+    if (G%mask2dT(i,j) > 0. .and. mass_shelf(i,j) .gt. 0.) then
```

Comment:
Please use '>' instead of '.gt.' in this routine, or at the very least do not mix this syntax on the same line!

Also, there should be spaced around the = when assigning variables, although these spaces can (and often are) omitted in do loops.  Please see the discussion on white space at https://github.com/NOAA-GFDL/MOM6/wiki/Code-style-guide.